RecoLocalTracker/SiStripZeroSuppression/plugins/SiStripMergeZeroSuppression.cc

```diff
@@ -0,0 +1,146 @@
+  m_zsDigisToMerge = consumes<edm::DetSetVector<SiStripDigi>>(conf.getParameter<edm::InputTag>("DigisToMergeZS"));
+  produces<edm::DetSetVector<SiStripDigi>>("ZeroSuppressed");
+}
+
+void SiStripMergeZeroSuppression::produce(edm::Event& event, const edm::EventSetup& /*eventSetup*/)
+{
+  std::cout<< "starting Merging" << std::endl;
+  edm::Handle<edm::DetSetVector<SiStripDigi>> inputdigi;
+  edm::Handle<edm::DetSetVector<SiStripRawDigi>> inputraw;
+  event.getByToken(m_rawDigisToMerge, inputdigi);
+  event.getByToken(m_zsDigisToMerge, inputraw);
+
+  std::cout << inputdigi->size() << " " << inputraw->size() << std::endl;
+  if ( ! inputraw->empty() ) {
+    std::vector<edm::DetSet<SiStripDigi>> outputdigi(inputdigi->begin(), inputdigi->end());
+
+    std::cout << "looping over the raw data collection" << std::endl;
+    for ( const auto& rawDigis : *inputraw ) {
+      edm::DetSet<SiStripRawDigi>::const_iterator itRawDigis = rawDigis.begin();
+      uint16_t nAPV = rawDigis.size()/128;
+      uint32_t rawDetId = rawDigis.id;
+
+      std::vector<bool> restoredAPV(std::size_t(nAPV), false);
+
+      bool isModuleRestored = false;
+      for ( uint16_t strip = 0; strip < rawDigis.size(); ++strip ) {
+        if(itRawDigis[strip].adc()!=0){
+          restoredAPV[strip/128] = true;
+          isModuleRestored = true;
+        }
+      }
+
+      if ( isModuleRestored ) {
+        std::cout << "apply the ZS to the raw data collection" << std::endl;
+        edm::DetSet<SiStripDigi> suppressedDigis(rawDetId);
+        m_algorithms->SuppressVirginRawData(rawDigis, suppressedDigis);
+
+        if ( ! suppressedDigis.empty() ) {
+          std::cout << "looking for the detId with the new ZS in the collection of the zero suppressed data" << std::endl;
+          bool isModuleInZscollection = false;
+          uint32_t zsDetId{0};
+          auto zsModule = std::lower_bound(std::begin(outputdigi), std::end(outputdigi), rawDetId,
+              [] ( const edm::DetSet<SiStripDigi>& elem, uint32_t testDetId ) { return elem.id < testDetId; });
+          if ( ( std::end(outputdigi) != zsModule ) && ( zsModule->id == rawDetId ) ) {
+            isModuleInZscollection = true;
+            zsDetId = zsModule->id;
+          }
+          std::cout << "after the look " << rawDetId << " ==== " <<  zsDetId << std::endl;
+          std::cout << "exiting looking for the detId with the new ZS in the collection of the zero suppressed data" << std::endl;
+
+          //creating the map containing the digis (in rawdigi format) merged
+          std::vector<uint16_t> MergedRawDigis(size_t(nAPV*128),0);
+
+          uint32_t count=0; // to be removed...
+          edm::DetSet<SiStripDigi> newDigiToIndert(rawDetId);
+          if ( ! isModuleInZscollection ) {
+            std::cout << "WE HAVE A PROBLEM, THE MODULE IS NTOT FOUND" << std::endl;
+            zsModule = outputdigi.insert(zsModule, newDigiToIndert);
```

Comment:
I'm not sure where this module is used (it was a boolean option `mergeCollections` of `SiStripZeroSuppression`, but there was almost no code overlap between the two cases so I split it out - I didn't find any config in cmssw that was using `mergeCollections=True`), so it's a bit hard to say. It looks like an earlier version of something like the hybrid ZS... Does anyone remember? Otherwise we may as well remove it.
In principle this should only happen when a module is in the VR but not in the ZS collection... I think the ZS algorithm saves an empty DetSet if there is nothing above baseline/noise, so in that case not that often.
Anyway, from the code above if this condition is true `zsModule` equals `outputdigi.end()`, so it's effectively a `push_back`